guides/developer/provider-setups/aws-dev-setup.md

```diff
@@ -0,0 +1,291 @@
+
+# Configure Development EC2 Instance
+
+Get your instance's DNS name and [login via SSH](https://docs.aws.amazon.com/quickstarts/latest/vmlaunch/step-2-connect-to-instance.html). 
+
+```
+ssh -A -L 9000:localhost:9000 -L 8084:localhost:8084 -L 8087:localhost:8087 ubuntu@$SPINNAKER_INSTANCE_DNS -i /path/to/my-key-pair.pem
+```
+
+Install dependencies:
+```
+curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.11/install.sh | bash
+source ~/.bashrc
+nvm install stable
+npm install -g yarn
+
+curl -O https://raw.githubusercontent.com/spinnaker/halyard/master/install/debian/InstallHalyard.sh
+sudo bash InstallHalyard.sh
+source ~/.bashrc
+hal -v
+```
+
+Configure and deploy the Spinnaker installation:
+```
+# Store state in S3 and deploy a recent stable version
+hal config storage s3 edit --region $AWS_REGION   # use region where your dev instance resides
+hal config storage edit --type s3
+hal config version edit --version 1.21.0  # or whichever is latest
+hal deploy apply
+
+sudo service apache2 stop
+sudo systemctl disable apache2
+
+# Workaround: https://github.com/spinnaker/spinnaker/issues/4041
+echo > ~/.hal/default/profiles/settings-local.js
+
+# Clone repos from your GitHub account
+hal config deploy edit --type localgit --git-origin-user={your github username}
+hal config version edit --version branch:master
+
+# Connect your AWS account
+## Replace the account id specified below (123456789012) with your own account id.
+hal config provider aws account add my-aws-devel-acct \
+    --account-id 123456789012 \
+    --assume-role role/SpinnakerManaged
+hal config provider aws account edit my-aws-devel-acct --regions $AWS_REGIONS  # regions you want Spinnaker to deploy to
+hal config provider aws enable
+
+# Connect your Docker registries
+hal config provider docker-registry enable
+
+hal config provider docker-registry account add my-dockerhub-devel-acct \
+    --address index.docker.io \
+    --repositories {your dockerhub username}/{your dockerhub repository} \
+    --username {your dockerhub username} \
+    --password \
+    --track-digests true
+
+# Replace the address specified below (123456789012.dkr.ecr.us-west-2.amazonaws.com) with the address to your own ecr registry.
+hal config provider docker-registry account add my-us-west-2-devel-registry \
+ --address 123456789012.dkr.ecr.us-west-2.amazonaws.com \
+ --username AWS \
+ --password-command "aws --region us-west-2 ecr get-authorization-token --output text --query 'authorizationData[].authorizationToken' | base64 -d | sed 's/^AWS://'" \
+ --track-digests true
+```
+
+Optionally connect Amazon ECS to your AWS account
+```
+hal config provider ecs account add ecs-my-aws-devel-acct --aws-account my-aws-devel-acct
+hal config provider ecs enable
+```
+
+Deploy everything
+```
+hal deploy apply
+```
+
+Wait for Clouddriver to start up by checking the logs in ~/dev/spinnaker/logs/clouddriver.log.
```

Comment:
Maybe include a line or two showing what will be in the log when Clouddriver is started?